src/transformers/modeling_utils.py

```diff
@@ -2976,7 +2974,7 @@ def from_pretrained(
                 loaded_state_dict_keys = sharded_metadata["all_checkpoint_keys"]
             else:
                 loaded_state_dict_keys = list(state_dict.keys())
-            if low_cpu_mem_usage or use_keep_in_fp32_modules:
+            if low_cpu_mem_usage or (use_keep_in_fp32_modules and is_accelerate_available()):
                 state_dict = None
```

Comment:
It is because of this controlflow: https://github.com/huggingface/transformers/blob/f29fe7458953dbf00addaf793d95ea1965bc8441/src/transformers/modeling_utils.py#L3541-L3553. For some reason the `else` branch uses [set_module_tensor_to_device](https://github.com/huggingface/transformers/blob/f29fe7458953dbf00addaf793d95ea1965bc8441/src/transformers/modeling_utils.py#L3619) that comes from accelerate. I just added the `or (use_keep_in_fp32_modules and is_accelerate_available())` to continue taking the same path as before in case `accelerate` is installed.

Unfortunately there are not many comments in `from_pretrained` to explain the different logic so it is hard to understand for me as well.